services/duty.service.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  validateDutyExists,
+  validateDutyIsModifiable,
+} from './validation/duty_validation';
+import {
+  Duty,
+  DutyQuery,
+  DutyQuerySchema,
+} from '../services/schemas/duty.zschema';
+import {
+  insertDuty,
+  getDutyByID,
+  getDutiesByQuery,
+  deleteDutyByID,
+  updateDutyByID,
+} from '../repositories/duty.repository';
+
+const createDuty = async (duty: Duty) => {
+  const result = await insertDuty({
+    ...duty,
+    soldiers: duty.soldiers ? duty.soldiers : [],
+  });
+  return result;
+};
+
+const findDutyByID = async (id: string) => {
+  const result = await getDutyByID(id);
+
+  validateDutyExists(id, result);
+
+  return result;
+};
+
+const findDutiesByQuery = async (query: DutyQuery) => {
+  const result = await getDutiesByQuery(query);
+  return result;
+};
+
+const deleteDuty = async (id: string) => {
+  const result = await getDutyByID(id);
+
+  validateDutyExists(id, result);
+
+  validateDutyIsModifiable(id, result);
+
+  await deleteDutyByID(id);
+  return result;
+};
+
+const updateDuty = async (id: string, update: DutyQuery) => {
+  const findResult = await getDutyByID(id);
+
+  validateDutyExists(id, findResult);
+
+  validateDutyIsModifiable(id, findResult);
+
+  if (Object.keys(DutyQuerySchema.parse(update)).length === 0) {
```

Comment:
No need to parse

Reply:
there is - I want to filter out unnecessary fields

Reply:
make schema strict, return 400 if filter is empty or only nonexistent fields